crates/bevy_text/src/font_atlas_set.rs

```diff
@@ -1,62 +1,61 @@
-use crate::{Font, FontAtlas};
-use ab_glyph::{Glyph, ScaleFont};
+use crate::{error::TextError, Font, FontAtlas};
+use ab_glyph::{GlyphId, OutlinedGlyph};
 use bevy_asset::{Assets, Handle};
 use bevy_core::FloatOrd;
 use bevy_math::Vec2;
 use bevy_render::texture::Texture;
 use bevy_sprite::TextureAtlas;
 use bevy_type_registry::TypeUuid;
-use bevy_utils::HashMap;
+use bevy_utils::{HashMap, HashMapExt};
 
-// work around rust's f32 order/hash limitations
 type FontSizeKey = FloatOrd;
 
-#[derive(Default, TypeUuid)]
+#[derive(TypeUuid)]
 #[uuid = "73ba778b-b6b5-4f45-982d-d21b6b86ace2"]
 pub struct FontAtlasSet {
-    font: Handle<Font>,
     font_atlases: HashMap<FontSizeKey, Vec<FontAtlas>>,
 }
 
-#[derive(Debug)]
+#[derive(Debug, Clone)]
 pub struct GlyphAtlasInfo {
     pub texture_atlas: Handle<TextureAtlas>,
-    pub char_index: u32,
+    pub glyph_index: u32,
 }
 
-impl FontAtlasSet {
-    pub fn new(font: Handle<Font>) -> Self {
-        Self {
-            font,
-            font_atlases: HashMap::default(),
+impl Default for FontAtlasSet {
+    fn default() -> Self {
+        FontAtlasSet {
+            font_atlases: HashMap::with_capacity(1),
         }
     }
+}
+
+impl FontAtlasSet {
+    pub fn new() -> Self {
```

Comment:
lets just remove `new()` here in favor of `default()`

Reply:
I removed `new` since you're in charge of the style of this codebase so I'm assuming that we should keep it symetrical with other `bevy` APIs. I would like ti mention this though : 

https://rust-lang.github.io/api-guidelines/interoperability.html?highlight=expected#types-eagerly-implement-common-traits-c-common-traits

> Note that it is common and expected for types to implement both Default and an empty new constructor. new is the constructor convention in Rust, and users expect it to exist, so if it is reasonable for the basic constructor to take no arguments, then it should, even if it is functionally identical to default.

Reply:
This is one of the few places where my opinions diverge from the api guidelines. I prefer "default only" because:

* it means theres "one way" to instantiate the type (a good thing ... lowers cognitive load)
* less code (easier to read, easier to maintain, _marginally_ faster to compile)